app/views/orders/index.html.haml

```diff
@@ -6,7 +6,7 @@
       = t '.new_order'
       %span.caret
     %ul.dropdown-menu
-      - Supplier.undeleted.order('suppliers.name ASC').each do |supplier|
+      - Supplier.where(id: Article.undeleted.select(:supplier_id).distinct).order('suppliers.name ASC').each do |supplier|
```

Comment:
Putting these queries in the view is bad practice. I'll move it to the controller when merged.